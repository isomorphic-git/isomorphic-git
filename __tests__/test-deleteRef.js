/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
const snapshots = require('./__snapshots__/test-deleteRef.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')
const { plugins, deleteRef, listTags } = require('isomorphic-git')

describe('deleteRef', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('deletes the latest tag ref to HEAD', async () => {
    // Setup
    let { fs, gitdir } = await makeFixture('test-deleteRef')
    plugins.set('fs', fs)
    // Test
    await deleteRef({
      gitdir,
      ref: 'refs/tags/latest'
    })
    let refs = await listTags({
      gitdir
    })
    expect(refs).toMatchSnapshot()
  })
})
