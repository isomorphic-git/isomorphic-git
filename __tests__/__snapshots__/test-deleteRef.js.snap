// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteRef deletes the latest tag ref to HEAD 1`] = `
Array [
  "prev",
]
`;
